src/controllers/user-data-manager.js

```diff
@@ -15,208 +21,320 @@ export default class UserDataManager {
-   *                        params: query parameters to
-   *                      }
-   * @return {Object[]} an array of data items
+   * Promise-based method - queries all objects from the wordlist by languageCode , only for only one wordItem
+   * or one wordItem from local/remote storage 
+   * @param {Object} data
+   * @param {String} data.languageCode - for quering all wordItems from wordList by languageCode
+   * @param {WordItem} data.wordItem - for quering one wordItem
+   * @param {Object} [params={ source: both, type: short, syncDelete: false }] - additional parameters for updating, now there are the following:
+   *                  params.source = [local, remote, both]
+   *                  params.type = [short, full] - short - short data for homonym, full - homonym with definitions data
+   * 
+   * @return {WordItem[]} 
    */
-  async query(data) {
-    // query queries both the remote and local stores and merges
-    // the results
-    let remoteAdapter =  this._remoteStorageAdapter(data.dataType)
-    let localAdapter = this._localStorageAdapter(data.dataType)
-
-    let remoteDataItems = await remoteAdapter.query(data.params)
-    let localDataItems = await localAdapter.query(data.params)
-
-    this.printErrors(localAdapter)
-
-    // if we have any remoteData items then we are going to
-    // reset the local store from the remoteData, adding back in any
-    // items that appeared only in the local
-    if (remoteDataItems.length > 0) {
-        localAdapter.deleteMany(params)
-    }
-    let addToRemote = []
-    let updateInRemote = []
-    
-    localDataItems.forEach(item => {
-      let inRemote = false
-      for (let i=0; i<remoteDataItems.length; i++ ) {
-        if (remoteDataItems[i].isSameItem(item)) {
-          inRemote = true
-          // if the item exists in the remote db, check to see if they differ
-          // and if so merge and update
-          if (remoteDataItems[i].isNotEqual(item)) {
-            let merged = remoteDataItems[i].merge(item)
-            remoteDataItems[i] = merged
-            updateInRemote.push(remoteDataItems[i].merge(item))
+  async query (data, params = {}) {
+    try {
+      params.type = params.type||'short'
+      params.source = params.source||'both'
+      params.syncDelete = params.syncDelete||false
+
+      let remoteAdapter =  this._remoteStorageAdapter(data.dataType)
+      let localAdapter = this._localStorageAdapter(data.dataType)
+
+      let finalItems = []
+      let remoteItems
+
+      if (params.source === 'local') {
+        finalItems = await localAdapter.query(data.params)
+      } else if (params.source === 'remote') {
+        remoteItems = await remoteAdapter.query(data.params)
+        for(let remoteItem of remoteItems) {
+          finalItems.push(localAdapter.dbDriver.createFromRemoteData(remoteItem))
+        }
+      } else {
+        remoteItems = await remoteAdapter.query(data.params)
+        if (params.type === 'full') {
```

Comment:
is 'full' used only in tests now?

Reply:
It is used to get full data about Homonym from UIController.
```
async onWordItemSelected (wordItem) {
    let wordItemFull = await this.userDataManager.query({ dataType: 'WordItem', params: { wordItem } }, { type: 'full' })
```